### PR TITLE
Custom first name / last name key for OpenID

### PR DESCRIPTION
**Pull request by xxsuperdreamxx**
#19827

Adding the ability to use a dynamic First and Last name for OpenID providers

### CODE REVIEW

File: api/src/auth/drivers/openid.ts
Comment (rijkvanzanten) [inline api/src/auth/drivers/openid.ts:177]:
```suggestion
			first_name: userInfo[this.config['firstNameKey'] ?? 'given_name'],
			last_name: userInfo[this.config['lastNameKey'] ?? 'family_name'],
```

**Review by rijkvanzanten — COMMENTED** [review]
opinions opinions

File: api/src/auth/drivers/openid.ts
Comment (xxsuperdreamxx) [reply api/src/auth/drivers/openid.ts:177]: I forgot that the double `??` existed. It is so confusing switching between multiple languages haha

**Comment by paescuj** [comment]
@xxsuperdreamxx Thank you! Would you mind updating the documentation (https://github.com/directus/directus/blob/main/docs/self-hosted/config-options.md#openid) too?

**Comment by xxsuperdreamxx** [comment]
@rijkvanzanten, most certainly, give me a moment

**Comment by paescuj** [comment]
> @rijkvanzanten, most certainly, give me a moment

Cool, thanks!
Before committing you can then run `pnpm -w format --write` to auto-fix the formatting issues 😃

**Comment by xxsuperdreamxx** [comment]
😂 Alrighty haha

**Comment by xxsuperdreamxx** [comment]
give me one moment Ziggo is being slow haha

**Comment by xxsuperdreamxx** [comment]
I used the VS code online editing feature because the Dutch are well-known for their internet issues, but we are way better than the Germans 😂

**Review by paescuj — APPROVED** [review]
LGTM! Thank you @xxsuperdreamxx for your contribution!

**Review by paescuj — CHANGES_REQUESTED** [review]
Waiting for feedback in https://github.com/directus/directus/issues/19827#issuecomment-1748450466 before merging 🚫

**Comment by xxsuperdreamxx** [comment]
Always happy to help! @paescuj

**Review by br41nslug — APPROVED** [review]

**Comment by paescuj** [comment]
Closing this for the reason outlined in https://github.com/directus/directus/issues/19827#issuecomment-1750442179.
Thank you anyway, @xxsuperdreamxx! ❤️ Looking forward to your next contribution 🚀

**Pull request closed without merging**
